### ISSUE TITLE
My node stopped making blocks

### ISSUE THREAD

**Issue by HM999** [input-output-hk/shelley-testnet#11]
**Describe the bug**
It stopped making blocks. log attached.

**To Reproduce**
https://github.com/input-output-hk/shelley-testnet/issues/10#issuecomment-504607074

**Expected behavior**
To continue making blocks

[genesis.log](https://github.com/input-output-hk/shelley-testnet/files/3316765/genesis.log)

**Comment by cloudy-brain** [comment]
Same here.

![Screenshot from 2019-06-22 13-18-44](https://user-images.githubusercontent.com/8825017/59963809-4ad44d00-94f0-11e9-9a43-b3f2bd3d80ba.png)

**Comment by HM999** [comment]
Is there some kind of debug / extra verbose way we should be running it?

**Comment by mzabaluev** [comment]
@HM999 Command line option `-vv` enables debug level messages, as does setting the `verbosity` field in the logger section of the config to 2 and above.

**Comment by phillewis** [comment]
I have also noticed this issue for some time now, even prior to the 0.1.0 release.  I have noticed it on x86_64-unknown-linux-musl builds (both self compiled and with a downloaded release) and have recently seen it on Mac OS (x86_64-apple-darwin) with the 0.2.3 release.

Sometimes when I start up I see nothing but message after message of end of epoch (following the initial startup messages), e.g.
...
{"msg":"End of epoch","level":"INFO","ts":"2019-06-24T16:56:57.004822+08:00","sub_task":"End Of Epoch Reminder","task":"leadership","epoch":57}
{"msg":"End of epoch","level":"INFO","ts":"2019-06-24T16:57:22.005368+08:00","sub_task":"End Of Epoch Reminder","task":"leadership","epoch":58}
{"msg":"End of epoch","level":"INFO","ts":"2019-06-24T16:57:47.004165+08:00","sub_task":"End Of Epoch Reminder","task":"leadership","epoch":59}
{"msg":"End of epoch","level":"INFO","ts":"2019-06-24T16:58:12.004462+08:00","sub_task":"End Of Epoch Reminder","task":"leadership","epoch":60}
{"msg":"End of epoch","level":"INFO","ts":"2019-06-24T16:58:37.003257+08:00","sub_task":"End Of Epoch Reminder","task":"leadership","epoch":61}
{"msg":"End of epoch","level":"INFO","ts":"2019-06-24T16:59:02.000858+08:00","sub_task":"End Of Epoch Reminder","task":"leadership","epoch":62}
{"msg":"End of epoch","level":"INFO","ts":"2019-06-24T16:59:27.000786+08:00","sub_task":"End Of Epoch Reminder","task":"leadership","epoch":63}
{"msg":"End of epoch","level":"INFO","ts":"2019-06-24T16:59:52.003206+08:00","sub_task":"End Of Epoch Reminder","task":"leadership","epoch":64}
{"msg":"End of epoch","level":"INFO","ts":"2019-06-24T17:00:17.005291+08:00","sub_task":"End Of Epoch Reminder","task":"leadership","epoch":65}
{"msg":"End of epoch","level":"INFO","ts":"2019-06-24T17:00:42.004353+08:00","sub_task":"End Of Epoch Reminder","task":"leadership","epoch":66}
{"msg":"End of epoch","level":"INFO","ts":"2019-06-24T17:01:07.001779+08:00","sub_task":"End Of Epoch Reminder","task":"leadership","epoch":67}
{"msg":"End of epoch","level":"INFO","ts":"2019-06-24T17:01:32.004336+08:00","sub_task":"End Of Epoch Reminder","task":"leadership","epoch":68}
{"msg":"End of epoch","level":"INFO","ts":"2019-06-24T17:01:57.005113+08:00","sub_task":"End Of Epoch Reminder","task":"leadership","epoch":69}
{"msg":"End of epoch","level":"INFO","ts":"2019-06-24T17:02:22.002049+08:00","sub_task":"End Of Epoch Reminder","task":"leadership","epoch":70}
....

I would say only about 1 in 5 will start up and process a transaction and even then as others have identified in other issues reports, it will often not process future transaction after processing the first 1 or 2.  There appears to be an underlying issue where a transaction will just stay Pending indefinitely.

If this is not being observed internally then could it be a platform issue?  How many OSes has it been tested on internally?

**Comment by phillewis** [comment]
Another piece of information that may be useful is that I have only ever used "genesis" for consensus and my genesis block has always been created with 2 UTXO addresses that have 50% each of the total value.

Only 1 of them is used to delegate and create the stake pool, so this means that the stake pool only has 50% of the total available value to stake with at the start.  This may explain why I can go long periods of time without seeing anything.

Others reporting a similar problem may be experiencing delays after some of the stake pool's value is transferred to another account that isn't staking.  At the end of the epoch the stake pool's stake would be seen as smaller.

If this is the cause then it would suggest that the node is not taking into consideration that while it may only have 50% (or another percentage for others) of the total value in the blockchain staked to it, this still constitutes 100% of the total value being staked, as there are no other stake pools.  It should therefore always be the node selected to process blocks.

**Comment by gmanroney** [comment]
This is duplicated in  #13

**Comment by mzabaluev** [comment]
Was your setup like the one described in input-output-hk/jormungandr#576?

**Comment by HM999** [comment]
Yes, I think it happened in that same sequence.

**Comment by dorin100** [comment]
@HM999 - is it ok to close this issue and track only https://github.com/input-output-hk/jormungandr/issues/576 ?

**Comment by HM999** [comment]
I mean thinking about it, might not even be a bug exactly - when does it decide which node will do which slots? Clearly it makes blocks to start with, i.e. when it has coins on the Faucet. But when I transfer them to an address but before they are delegated -- should the node be making blocks? Maybe not. Although if that is the case, on a network with only one node, how would further changes to the blockchain be made...

Basically, I'm not sure exactly how it is meant to work, is there a spec to test against, let me know and I will read it.

But, sure, close this one, sounds the same thing.

**Comment by dorin100** [comment]
Ok. Closing this one. Thank you @HM999 .

Please check @vincenthz 's comment from [here](https://github.com/input-output-hk/shelley-testnet/issues/17#issuecomment-505017314) for some more details about this corner case.

**Comment by phillewis** [comment]
Doesn't matter which issue is used to track this, but it sounds like there have been a few similar reports.  My comment above refers to a similar issue that can happen even after initial startup.

I am thinking that the application may be assuming there are other nodes as it doesn't have all value, even if it has all the value that is staked.
